src/thor/bidirectional_astar.cc

```diff
@@ -137,6 +137,10 @@ bool BidirectionalAStar::ExpandForward(GraphReader& graphreader,
   if (!costing_->Allowed(nodeinfo)) {
     const DirectedEdge* opp_edge;
     const GraphId opp_edge_id = graphreader.GetOpposingEdgeId(pred.edgeid(), opp_edge, tile);
+    // Check if edge is null before using it (can happen with regional data sets)
+    if (!opp_edge) {
+      return false;
+    }
     EdgeStatusInfo* opp_status = edgestatus_forward_.GetPtr(opp_edge_id, tile);
     return ExpandForwardInner(graphreader, pred, nodeinfo, pred_idx,
                               {opp_edge, opp_edge_id, opp_status}, shortcuts, tile, offset_time);
```

Comment:
you can avoid the branch (havent checked the peformance difference myself) by doing this:

```suggestion
    // Check if edge is null before using it (can happen with regional data sets)
    return opp_edge && ExpandForwardInner(graphreader, pred, nodeinfo, pred_idx,
       {opp_edge, opp_edge_id, edgestatus_forward_.GetPtr(opp_edge_id, tile)}, shortcuts, tile, offset_time);
```